### PR TITLE
feat(earn): wisdomtree catalogue client + deposit-eligibility capability

### PR DESCRIPTION
**Pull request by cutbow7**
Implements the SDK-free half of the WisdomTree integration in @sdp/earn:

- providers/wisdomtree/connect.ts — the Connect REST client (OAuth2 password grant with a per-credential token cache, products, on-receipt wallets, organization wallets, an orders tooling read). Wire shapes come from WisdomTree's published OpenAPI spec; SDP holds no credentials yet, so every field the spec leaves unprinted is marked UNVERIFIED and read through one constant or accessor so the live-measurement fix is one edit.
- listStrategies — production only (WisdomTree is Solana-mainnet-only; other environments answer [] and get browse rows via the PRO-1742 mirror). Identity comes from the measured on-chain registry in @sdp/types; availability comes from the products API, fail-closed on can_trade !== true; a malformed response throws so the sync skips the pass rather than delisting the shelf. No currentApy on purpose — the 7-day yield lives in a Fund Data credential SDP does not hold, and a missing rate beats a fabricated one.
- EarnDepositEligibilityProvider — a NEW optional capability: a provider- side eligibility check money-IN paths consult (never exits, per ADR 0002). Exists for regulated instruments whose transfer hook would fail settlement to an unverified wallet: the check turns 'your USDC left and nothing came back' into a refusal with WisdomTree's reason attached, BEFORE money moves. WisdomTree implements it over the Connect wallet registry, fail-closed on unregistered/unapproved/unknown statuses.

### CODE REVIEW

File: packages/sdp-earn/src/providers/wisdomtree/connect.ts [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+
+  let parsed: unknown;
+  try {
+    parsed = JSON.parse(raw);
+  } catch {
+    throw providerNotConfigured(
+      `${keyName} is not valid JSON. Expected {"clientId","clientSecret","username","password"}.`
+    );
+  }
+  const record = parsed as Partial<Record<(typeof CREDENTIAL_FIELDS)[number], unknown>>;
```
Comment (keeganthomp) [inline packages/sdp-earn/src/providers/wisdomtree/connect.ts:93]: [P2] Reject non-object credential JSON cleanly

`JSON.parse("null")` succeeds, then `record[field]` throws a raw TypeError instead of the promised `PROVIDER_NOT_CONFIGURED` error. Guard that the parsed value is a non-null, non-array object before indexing so malformed packed credentials stay a classified configuration failure.

**Review by keeganthomp — DISMISSED** [review]
No blocking issue is introduced by this delta. The stack still inherits #1535's provider-availability test failure; the parser edge case and existing review threads are non-blocking.
